api/src/decorators/validate-units-required.decorator.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidationArguments,
+} from 'class-validator';
+import { PrismaService } from '../services/prisma.service';
+import { ListingPublishedUpdate } from '../dtos/listings/listing-published-update.dto';
+import { FeatureFlagEnum } from '../enums/feature-flags/feature-flags-enum';
+
+export function ValidateUnitsRequired(validationOptions?: ValidationOptions) {
```

Comment:
💯 love this decorator
nit: Can you add a comment above this so future people know why we have this function and why it's only required if `enableUnitGroups` is false